orchestrator-salt/src/main/resources/salt/salt/cloudera/agent/init.sls

```diff
@@ -24,7 +24,17 @@ install-cloudera-manager-agent:
 
 {% endif %}
 
-{%- if not salt['pkg.version']('python-psycopg2') and not salt['pkg.version']('python2-psycopg2') and not salt['pkg.version']('python38-psycopg2') %}
+# The package we check for is called differently for different OS + Python version 
+# combinations and we need to be careful to keep backward compatibility.
+# 1. CentOS 7 + Python 2
+# 2. RedHat 7 + Python 2 (customer specific)
+# 3. RedHat 8 + Python 3.8
+# 4. CentOS 8 + Python 3.8
+{%- if not salt['pkg.version']('python-psycopg2')
```

Comment:
The cmd.run also has an _unless_ condition, do we really need this explicit check?

Reply:
Using "unless" here looks sub-optimal and complicated. According to its documentation:

"The unless requisite specifies that a state should only run when any of the specified commands return False. "

which means it only works with shell commands. In turn, that would mean that instead of conveniently letting SaltStack do the heavy lifting by calling dnf/yum/etc. under the hood depending on the OS, I should call those manually, then grep the results, etc.